### PR TITLE
Querying GraphQL service with bearer token

### PR DESCRIPTION
**Pull request by MChartier**
Client-side changes to query GraphQL service with JWT access token included in Authorization header.

TODO: Rebase onto Olga's typed GraphQL client once available.
TODO: On first sign in of new user, create new user record in database.

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -65,8 +63,21 @@ function App() {
     );
   }
 
+  const apolloClient = new ApolloClient({
+    uri: `${apiUrl}/query`,
+    request: (operation) => {
+      const token = user.AccessToken;
```
Comment (SlyCaptainFlint) [inline src/App.tsx:69]: what is `user` in this context? where is it defined?

File: src/utils/auth/auth.ts [hunk 2]
```diff
@@ -44,9 +44,21 @@ export default class Auth {
...
+    const redirectUrl = AuthSession.getRedirectUrl();
+    const authQueryParams: any = {
+      response_type: 'token',
+      client_id: auth0ClientId,
+      redirect_uri: redirectUrl,
+      prompt: 'login',
+      scope: 'openid profile',
+      audience: 'matterhorn-api'
+    };
+    const authQueryParamString = Object.keys(authQueryParams)
```
Comment (SlyCaptainFlint) [inline src/utils/auth/auth.ts:56]: to simplify here, I think you could use Object.entries, which would then pass both the key and value as params to the map function

File: src/views/UserScreen.tsx [hunk 3]
```diff
@@ -12,6 +14,21 @@ export interface UserScreenProps {
 export default function UserScreen(props: UserScreenProps) {
   const { user } = props;
 
+  const { data, loading, refetch } = useQuery(ME_QUERY, {
+    notifyOnNetworkStatusChange: true,
+    context: {
```
Comment (SlyCaptainFlint) [inline src/views/UserScreen.tsx:19]: does this need to be defined here in addition to the more general case you seem to have set up when you created the apollo client in App.tsx?

File: src/views/UserScreen.tsx [hunk 4]
```diff
@@ -12,6 +14,21 @@ export interface UserScreenProps {
...
+      headers: {
+        Authorization: `Bearer ${user.AccessToken}`
+      }
+    },
+    onError: ((error: any) => {
```
Comment (SlyCaptainFlint) [inline src/views/UserScreen.tsx:24]: I think we could probably remove a lot of this console.log logic before checking in

File: src/App.tsx (hunk 1 above)
Comment (MChartier) [reply src/App.tsx:69]: It is a state variable declared on line 29. It is set by a call to `setUser` in `signIn` on line 39.

**Pull request merged**